ckanext/dcatapit/model/dcatapit_model.py

```diff
@@ -93,4 +94,134 @@ def persist(self, tag, lang):
             log.error('Exception occurred while persisting DB objects: %s', e)
             raise
 
-meta.mapper(DCATAPITTagVocabulary, dcatapit_vocabulary_table)
\ No newline at end of file
+
+meta.mapper(DCATAPITTagVocabulary, dcatapit_vocabulary_table)
+
+MAPPING_SECTION = 'dcatapit:theme_group_mapping'
+DCATAPIT_THEME_TO_MAPPING_SOURCE = 'ckanext.dcatapit.theme_group_mapping.file'
+
+
+def get_theme_to_groups():
+    """
+    Returns dictionary with groups for themes
+    """
+    fname = config.get(DCATAPIT_THEME_TO_MAPPING_SOURCE)
+    if not fname or not os.path.exists(fname):
+        log.warning("Cannot parse theme mapping, no such file: %s", fname)
+        return
+    return import_theme_to_group(fname)
+
+
+def _clean_groups(package):
+    """
+    Clears package's groups
+    """
+    Session.query(Member).filter(Member.table_name == 'package',
+                                 Member.table_id == package.id,
+                                 Member.capacity != 'admin')\
+                         .update({'state':'deleted'})
+
+
+def _add_groups(package, groups):
+    """
+    Adds groups to package
+    """
+    for g in groups:
+        member = Member(state='active',
+                        table_id=package.id,
+                        group_id=g.id,
+                        table_name='package')
+        Session.add(member)
+
+def _get_group_from_session(gname):
+    """
+    If Group was created within current session, get
+    it from cache instead of db.
+
+    This exists because new, uncommited/unflushed objects are 
+    not accessible by Session.query.
+    """
+    for obj in Session.new:
+        if isinstance(obj, Group):
+            if obj.name == gname:
+                return obj
+
+def populate_theme_groups(instance, clean_existing=False):
+    """
+    For given instance, it finds groups from mapping corresponding to
+    Dataset's themes, and will assign dataset to those groups.
+
+    Existing groups will be removed, if clean_existing is set to True.
+
+    If group doesn't exist, it will be created.
+    """
+    themes = instance.extras.get('theme')
+    if not themes:
+        log.debug("no theme from %s", instance)
+        return
+    theme_map = get_theme_to_groups()
+    if not theme_map:
+        log.warning("Theme to group map is empty")
+        return
+    if not isinstance(themes, list):
+        themes = [themes]
+    all_groups = set()
+    for theme in themes:
+        _groups = theme_map.get(theme)
+        if not _groups:
+            continue
+        all_groups = all_groups.union(set(_groups))
+
+    if clean_existing:
+        _clean_groups(instance)
+    groups = []
+    for gname in all_groups:
+        group =  Session.query(Group)\
+                        .filter(Group.name==gname)\
+                        .first() or _get_group_from_session(gname)
+        if group is None:
+            group = Group(name=gname)
+            Session.add(group)

```

Comment:
Please consider adding an option like `ckanext.dcatapit.theme_group_mapping.use_group` with allowed values:
- `only existing`: add the mapping only if the group already exists
- `force_create`: create the group if it doesn't exist

If the option isn't in the config file, the behaviour should be the most conservative one (`only existing` in this case)